### PR TITLE
fix: add JSON schema file

### LINKED ISSUE #5: JSON Schema File
### Description

Would it be possible to host a [JSON schema](https://json-schema.org) for the `cz.json` configuration file in order to enable the syntax checks? 

For example, in Deno projects we add the following field to `deno.json` file:

```
"$schema": "https://deno.land/x/deno/cli/schemas/config-file.v1.json"
```

This could be extremely useful for configuration checks and validation.

### Possible Solution

_No response_

### Additional context

_No response_

### Related issues

_No response_

### PR DESCRIPTION
**Pull request by HollowManTest**
## Description

add a [JSON schema](https://json-schema.org/) for the cz.json configuration file in order to enable the syntax checks?

For example, in Deno projects we add the following field to deno.json file:

"$schema": "https://deno.land/x/deno/cli/schemas/config-file.v1.json"
This could be extremely useful for configuration checks and validation.

## Checklist

- [X] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [X] Add test cases to all the changes you introduce
- [X] Run `uv run poe all` locally to ensure this change passes linter check and tests
- [X] Manually test the changes:
  - [X] Verify the feature/bug fix works as expected in real-world scenarios
  - [X] Test edge cases and error conditions
  - [X] Ensure backward compatibility is maintained
  - [X] Document any manual testing steps performed
- [X] Update the documentation for the changes

### Documentation Changes

- [X] Run `uv run poe doc` locally to ensure the documentation pages renders correctly
- [X] Check and fix any broken links (internal or external) in the documentation

> When running `uv run poe doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

## Expected Behavior

JSON schema works correctly.

## Steps to Test This Pull Request

Run test cases

## Additional Context

Resolves https://github.com/swe-productivity/commitizen/issues/5

### CODE REVIEW

**Review by bearomorphism — CHANGES_REQUESTED** [review]
I need detailed manual verification steps, for example, screenshots that the schema validation indeed works in the IDEs

File: schemas/test_schema.py
[the remark this answers is not in the document]
Comment (HollowManTest) [reply schemas/test_schema.py]: Now switched to pytest

File: tests/test_json_schema.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+            ({"commitizen": {"pre_bump_hooks": "not-an-array"}}, "pre_bump_hooks_not_array"),
+            # Invalid type for post_bump_hooks
+            ({"commitizen": {"post_bump_hooks": 123}}, "post_bump_hooks_not_array"),
+            # Invalid style element (not 2 items)
+            ({"commitizen": {"style": [["qmark"]]}}, "style_element_wrong_length"),
+            # Invalid type for prerelease_offset
+            ({"commitizen": {"prerelease_offset": "zero"}}, "prerelease_offset_not_number"),
+            # Invalid type for message_length_limit
+            ({"commitizen": {"message_length_limit": "100"}}, "message_length_limit_not_number"),
+        ],
```
Comment (bearomorphism) [inline tests/test_json_schema.py]: Use pytest.param and make those comments as ids

**Review by bearomorphism — COMMENTED** [review]
some nits

**Review by bearomorphism — APPROVED** [review]
LGTM, thanks!
